test/UniswapV3Staker.unit.spec.ts

```diff
@@ -397,14 +402,113 @@ describe('UniswapV3Staker.unit', async () => {
+
+        createIncentiveResult = await helpers.createIncentiveFlow({
+          rewardToken: context.rewardToken,
+          poolAddress: context.poolObj.address,
+          startTime,
+          totalReward,
+        })
+
+        await Time.setAndMine(startTime + 1)
+
+        // Make sure we're starting from a clean slate
+        const depositBefore = await context.staker.deposits(tokenId)
+        expect(depositBefore.owner).to.eq(constants.AddressZero)
+        expect(depositBefore.numberOfStakes).to.eq(0)
+
+        subject = (data: string, actor: Wallet = lpUser0) => {
+          return context.nft
+            .connect(actor)
+            [SAFE_TRANSFER_FROM_SIGNATURE](
+              actor.address,
+              context.staker.address,
+              tokenId,
+              data,
+              {
+                ...maxGas,
+                from: actor.address,
+              }
+            )
+        }
+      })
+
+      it('allows depositing without staking', async () => {
+        // Pass empty data
+        await subject(ethers.utils.defaultAbiCoder.encode([], []), lpUser0)
+        const deposit = await context.staker.deposits(tokenId)
+        expect(deposit.owner).to.eq(lpUser0.address)
+        // Make sure it is not staked.
+        expect(deposit.numberOfStakes).to.eq(BN('0'))
+      })
+
+      it('allows depositing and staking for a single incentive', async () => {
+        const data = ethers.utils.defaultAbiCoder.encode(
+          [INCENTIVE_KEY_ABI],
+          [incentiveResultToStakeAdapter(createIncentiveResult)]
+        )
+        await subject(data, lpUser0)
+        const deposit = await context.staker.deposits(tokenId)
+        expect(deposit.owner).to.eq(lpUser0.address)
+        expect(deposit.numberOfStakes).to.eq(BN('1'))
+      })
+
+      it('allows depositing and staking for two incentives', async () => {
+        const incentive2 = await helpers.createIncentiveFlow({
+          rewardToken: context.rewardToken,
+          poolAddress: context.poolObj.address,
+          startTime: createIncentiveResult.startTime + 100,
+          totalReward,
+        })
+
+        await Time.setAndMine(incentive2.startTime)
+
+        const data = ethers.utils.defaultAbiCoder.encode(
+          [`${INCENTIVE_KEY_ABI}[]`],
+          [
+            [createIncentiveResult, incentive2].map(
+              incentiveResultToStakeAdapter
+            ),
+          ]
+        )
+        await subject(data, lpUser0)
+        const deposit = await context.staker.deposits(tokenId)
+        expect(deposit.owner).to.eq(lpUser0.address)
+        expect(deposit.numberOfStakes).to.eq(BN('2'))
+      })
+
+      describe('reverts when', () => {
+        it('staking info is less than 160 bytes and greater than 0 bytes', async () => {
+          const data = ethers.utils.defaultAbiCoder.encode(
+            [INCENTIVE_KEY_ABI],
+            [
+              // Make the data invalid
+              incentiveResultToStakeAdapter({
```

Comment:
this is still a valid 160 byte payload, no? can do `.slice` on the encoded data to remove a byte

Reply:
Ah, when I do that, ethers throws an error. I can add another test to do a trimmed payload

Reply:
I kept this test and added the short payload one here: https://github.com/omarish/uniswap-v3-staker/pull/158/files#diff-992e2d1704b197612cb228120eb3f0fa924ee871f70a718b797b93e5cf6a2ff2R536